src/library-authoring/units/LibraryUnitBlocks.tsx

```diff
@@ -238,9 +238,7 @@ export const LibraryUnitBlocks = ({ preview }: LibraryUnitBlocksProps) => {
   const [hidePreviewFor, setHidePreviewFor] = useState<string | null>(null);
   const { showToast } = useContext(ToastContext);
 
-  const { readOnly, showOnlyPublished } = useLibraryContext();
-  const { sidebarComponentInfo } = useSidebarContext();
-  const unitId = sidebarComponentInfo?.id;
```

Comment:
Hi @navinkarkera!
Thanks for fixing that!

Just to note: we have some dubious behaviour on our `LibraryContext` that is causing bugs, and we will probably need to fix it soon.
In the context, we are using `collectionId/unitId` to indicate both that the `Collection/Unit Page` is open AND that the `collection/unit` is selected on the sidebar (instead of using the `sidebarComponentInfo`.

Reply:
@rpenido Yes, we need to come up with a cleaner solution to navigate i.e. update url and select appropriate component/collection/unit.
We could use `sidebarComponentInfo` to denote selection instead of `collectionId/componentId/unitId` in library context.